### PR TITLE
hasura graphql-engine: 1.0.0 -> 2.0.5

### PR DESCRIPTION
**Pull request by Lassulus**
###### Motivation for this change

update and fix hasura graphql-engine. Not sure how to fix odbc the right way, so it will persist and not be broken again with the next upgrade

###### Things done
Hasura

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by cdepillabout — CHANGES_REQUESTED** [review]
Thanks for taking a look at this.

There are a couple things that should be cleaned up here:

1. Can you rebase this on the `haskell-updates` branch?  In general we try to merge all Haskell-related changes into that branch.
2. I couldn't quite tell, but are the various files in `pkgs/development/misc/haskell/hasura/` generated with `cabal2nx`?  If that is the case, then they probably shouldn't have things manually changed with them.  For instance, I imagine you manually added a patch to `pkgs/development/misc/haskell/hasura/graphql-engine/default.nix` and set the `check = false;` line in `pkgs/development/misc/haskell/hasura/pg-client/default.nix`.  These types of things should instead be set in one of the files like `pkgs/development/haskell-modules/configuration-common.nix`.
3. Is the patch really necessary?  In general we try to not carry around patches in Nixpkgs if we can at all avoid it.  For instance, would it be possible to pull this patch from upstream?
3. Would you be able to ask upstream to just release Hasura to Hackage?  If they released to Hackage, then we wouldn't have to carry around all these files generated with `cabal2nix`, since we could just get everything from Hackage.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133621` at 8064cf77 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c073aa4ea73a1d67f3c03e466bc37c7e">hasura-cli</a></li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>haskellPackages.graphql-parser</li>
    <li>haskellPackages.odbc</li>
    <li>haskellPackages.pg-client</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/haskell-modules/generic-builder.nix:663:3:
```
    |
663 |   meta = { inherit homepage license platforms; }
    |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by cdepillabout** [comment]
> I also touched the versions and added myself to maintainers. should all of this happen in configuration-common.nix?

The way we normally do it is to have one file that is just the output of running `cabal2nix`, with no manual changes.  `pkgs/development/misc/haskell/hasura/graphql-engine/default.nix` might be similar to something like this.

Then this file is imported in `pkgs/development/haskell-modules/non-hackage-packages.nix`.  At this point the package becomes mostly a normal package in our Nixpkgs Haskell package set.

If you want to add maintainers or patches, you should do it in either `pkgs/development/haskell-modules/configuration-common.nix` or `configuration-nix.nix`, similar to the normal Haskell packages.

Some Haskell packages then have a separate file somewhere under `pkgs/` that makes other changes or does things like `justStaticExecutables`.

For instance, if you look at `spago`, it is basically setup like this.  It is not on Hackage, so a file generated with `cabal2nix` is in `pkgs/development/tools/purescript/spago/spago.nix`.  This is then imported in `pkgs/development/haskell-modules/non-hackage-packages.nix`.  Some things are fixed in `pkgs/development/haskell-modules/configuration-nix.nix`.  Then there is a file `pkgs/development/tools/purescript/spago/default.nix` that adds some more things (like tests, maintainers, etc).  This is the file that is then imported in `pkgs/top-level/all-packages.nix`.  I'd suggest trying to lay out hasura similarly.

Spago also has an update script at `pkgs/development/tools/purescript/spago/update.sh`.  This makes it easy for anyone to bump the spago version without having to figure out how to manually run `cabal2nix`.  If possible, I'd recommend you try to make something similar for Hasura.  Although not having an update script won't prevent this PR from being merged in.

> touched the versions

In general, the versions should be figured out when you run `cabal2nix`.  I'd be surprised if you had to update them manually.

> I don't know anyone upstream. the issue tracker seems pretty "full" But I can try to find an issue or create one if I can't find one.

If you could create an issue, that would be a big help.  It would at least let them know that it is harder for distros to package their software if they don't make releases to Hackage.

**Comment by Lassulus** [comment]
So, I did some stuff. sadly I was not able to get rid of the patch, so it will stay for now, until either a fix becomes upstream available or someone figures out whats actually wrong :D 
there was also a new release (2.0.6) but there were new build errors which I haven't figured out yet (maybe another dep needs to be vendored?)
Also the version of graphql-engine seems to be hardcoded to 1.0.0 in the upstream cabal file. Not sure how to continue there. Help and Reviews are appreciated

**Comment by cdepillabout** [comment]
This all looks good to me.

> Also the version of graphql-engine seems to be hardcoded to 1.0.0 in the upstream cabal file.

This is pretty unfortunate.  Not setting the version in the .cabal file is really bad practice.

Maybe you could send a PR fixing this for them?

Alternatively, you could work around this by manually setting the version:

```diff
diff --git a/pkgs/development/haskell-modules/configuration-common.nix b/pkgs/development/haskell-modules/configuration-common.nix
index ef8647af1d8..8795b0c9ffe 100644
--- a/pkgs/development/haskell-modules/configuration-common.nix
+++ b/pkgs/development/haskell-modules/configuration-common.nix
@@ -1297,10 +1297,11 @@ self: super: {
   graphql-engine = overrideCabal (super.graphql-engine.override {
     immortal = self.immortal_0_2_2_1;
   }) (drv: {
     patches = [ ./patches/graphql-engine-mapkeys.patch ];
     isLibrary = false;
+    version = "2.0.5";
   });
   pg-client = overrideCabal super.pg-client (drv: {
     librarySystemDepends = with pkgs; [ postgresql krb5.dev openssl.dev ];
     # wants a running DB to check against
     doCheck = false;
```

You could probably automate this with some `sed` / `awk` hackery in the update script you wrote (which is really convenient btw!).

-------------------------------

One other thing.  If you use hasura on a regular basis, I'd recommend marking yourself as a maintainer.  You'll then get pinged on the `haskell-update` PRs if hasura-graphql-engine ever stops building.

One way you could do this is something like:

```diff
diff --git a/pkgs/development/misc/haskell/hasura/update.sh b/pkgs/development/misc/haskell/hasura/update.sh
index cfb894ed844..749c569b64c 100755
--- a/pkgs/development/misc/haskell/hasura/update.sh
+++ b/pkgs/development/misc/haskell/hasura/update.sh
@@ -32,7 +32,7 @@ echo "# This has been automatically generated by the script" > "$engine_derivati
 echo "# ./update.sh.  This should not be changed by hand." >> "$engine_derivation_file"
 
 # cabal2nix --revision "$new_version" --subpath server "https://github.com/hasura/graphql-engine.git" >> "$engine_derivation_file"
-cabal2nix --revision "v2.0.5" --subpath server --no-check "https://github.com/hasura/graphql-engine.git" >> "$engine_derivation_file"
+cabal2nix --revision "v2.0.5" --subpath server --maintainer lassulus  --no-check "https://github.com/hasura/graphql-engine.git" >> "$engine_derivation_file"
 
 echo "Running cabal2nix and outputting to ${parser_derivation_file}..."
```

This will correctly add you as a maintainer to the graphql-engine nix file like:

```diff
diff --git a/pkgs/development/misc/haskell/hasura/graphql-engine.nix b/pkgs/development/misc/haskell/hasura/graphql-engine.nix
index 6af6bbd4262..285094874e9 100644
--- a/pkgs/development/misc/haskell/hasura/graphql-engine.nix
+++ b/pkgs/development/misc/haskell/hasura/graphql-engine.nix
@@ -77,4 +77,5 @@ mkDerivation {
   homepage = "https://www.hasura.io";
   description = "GraphQL API over Postgres";
   license = lib.licenses.asl20;
+  maintainers = with lib.maintainers; [ lassulus ];
 }
```

-----------------------------------------

Overall, this PR looks good to me and I'm happy to merge it in when you're ready.

File: pkgs/development/misc/haskell/hasura/update.sh
Comment (cdepillabout) [inline pkgs/development/misc/haskell/hasura/update.sh:35]: Could you add a comment in the code here explaining why `v2.0.5` is hard-coded on this line?  (And probably suggesting that it should be removed in the future)

**Comment by Lassulus** [comment]
Alright, did everything, so should be ready from my side.
After merging this I will open up a draft PR where I will post the errors of updating to 2.0.6 and what happens without the mapKeys patch.

File: pkgs/development/haskell-modules/configuration-nix.nix
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-nix.nix:228]: Try using `addExtraLibraries` instead of `addBuildDepends` which would avoid propagating.

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (sternenseemann) [inline pkgs/development/haskell-modules/configuration-common.nix:1301]: Maybe add a note here why this is needed? I was a bit confused looking at the code, but given that graphlql-engine is a server this shouldn't matter much.

Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: why the `isLibrary = false` ? actually I don't know, it just fails if I have it enabled

File: pkgs/development/haskell-modules/non-hackage-packages.nix
Comment (sternenseemann) [inline pkgs/development/haskell-modules/non-hackage-packages.nix:36]: `graphql-engine` also depends on `resource-pool` and there is a clashing one on hackage unfortunately. Maybe a comment warning about this fact here may be appropriate.

```suggestion
  graphql-engine = self.callPackage ../misc/haskell/hasura/graphql-engine.nix {
      resource-pool = self.callPackage ../misc/haskell/hasura/pool.nix {};
  };
```

**Review by sternenseemann — CHANGES_REQUESTED** [review]

File: pkgs/development/haskell-modules/non-hackage-packages.nix
Comment (Lassulus) [reply pkgs/development/haskell-modules/non-hackage-packages.nix:36]: the pool.nix is actually a fork of the upstream resource-pool.

Comment (sternenseemann) [reply pkgs/development/haskell-modules/non-hackage-packages.nix:36]: In any case we should use it since it is propagated from `pg-client` — conflicting packages tend to cause problems.

Comment (Lassulus) [reply pkgs/development/haskell-modules/non-hackage-packages.nix:36]: yep, testing it now, will push afterwards

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: This should be fixed by deleting all the [comments above this line](https://github.com/hasura/graphql-engine/blob/5990ca403bf6e7dd66081720c329e513f0624fb6/server/src-lib/Hasura/Backends/BigQuery/Types.hs#L385-L400) via a patch or similar, I think at least.

Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: ah, nice, while I have you here. Maybe you know whats wrong with this error: my patch fixes it, but I'm not sure it does make any sense (since the local defined mapKeys is now unused):
```
src-lib/Data/HashMap/Strict/Extended.hs:10:5: error:
    Ambiguous occurrence ‘mapKeys’
    It could refer to
       either ‘M.mapKeys’,
              imported from ‘Data.HashMap.Strict’ at src-lib/Data/HashMap/Strict/Extended.hs:20:1-41
              (and originally defined in ‘Data.HashMap.Internal’)
           or ‘Data.HashMap.Strict.Extended.mapKeys’,
              defined at src-lib/Data/HashMap/Strict/Extended.hs:69:1
   |
10 |   , mapKeys
   |     ^^^^^^^
   ```

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]:
```diff
- import           Data.HashMap.Strict as M
+ import qualified Data.HashMap.Strict as M
```

in that file should fix that without deleting the export.

Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: sadly it doesn't seem to be that easy:
```
src-lib/Data/HashMap/Strict/Extended.hs:25:14: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
25 | catMaybes :: HashMap k (Maybe v) -> HashMap k v
   |              ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:25:37: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
25 | catMaybes :: HashMap k (Maybe v) -> HashMap k v
   |                                     ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:28:56: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
28 | fromListOn :: (Eq k, Hashable k) => (v -> k) -> [v] -> HashMap k v
   |                                                        ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:32:51: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
32 | unionsAll :: (Eq k, Hashable k, Foldable t) => t (HashMap k v) -> HashMap k (NonEmpty v)
   |                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:32:67: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
32 | unionsAll :: (Eq k, Hashable k, Foldable t) => t (HashMap k v) -> HashMap k (NonEmpty v)
   |                                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:41:65: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
41 | groupOn :: (Eq k, Hashable k, Foldable t) => (v -> k) -> t v -> HashMap k [v]
   |                                                                 ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:44:67: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
44 | groupOnNE :: (Eq k, Hashable k, Foldable t) => (v -> k) -> t v -> HashMap k (NonEmpty v)
   |                                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:47:77: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
47 | differenceOn :: (Eq k, Hashable k, Foldable t) => (v -> k) -> t v -> t v -> HashMap k v
   |                                                                             ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:51:34: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
51 | lpadZip :: (Eq k, Hashable k) => HashMap k a -> HashMap k b -> HashMap k (Maybe a, b)
   |                                  ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:51:49: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
51 | lpadZip :: (Eq k, Hashable k) => HashMap k a -> HashMap k b -> HashMap k (Maybe a, b)
   |                                                 ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:51:64: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
51 | lpadZip :: (Eq k, Hashable k) => HashMap k a -> HashMap k b -> HashMap k (Maybe a, b)
   |                                                                ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:68:51: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
68 | mapKeys :: (Ord k2, Hashable k2) => (k1 -> k2) -> HashMap k1 a -> HashMap k2 a
   |                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:68:67: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
68 | mapKeys :: (Ord k2, Hashable k2) => (k1 -> k2) -> HashMap k1 a -> HashMap k2 a
   |                                                                   ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:78:70: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
78 | unionsWith :: (Foldable f, Hashable k, Ord k) => (a -> a -> a) -> f (HashMap k a) -> HashMap k a
   |                                                                      ^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:78:86: error:
    Not in scope: type constructor or class ‘HashMap’
    Perhaps you meant ‘M.HashMap’ (imported from Data.HashMap.Strict)
   |
78 | unionsWith :: (Foldable f, Hashable k, Ord k) => (a -> a -> a) -> f (HashMap k a) -> HashMap k a
   |                                                                                      ^^^^^^^
   ```

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: Then let's make it

```diff
- import           Data.HashMap.Strict as M
+ import qualified Data.HashMap.Strict as M
+ import           Data.HashMap.Strict (HashMap ())
```

Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: already had a similiar idea:
```
rc-lib/Data/HashMap/Strict/Extended.hs:30:16: error:
    • Variable not in scope: fromList :: [(k, v)] -> HashMap k v
    • Perhaps you meant one of these:
        ‘M.fromList’ (imported from Data.HashMap.Strict),
        ‘fromListOn’ (line 30)
      Perhaps you want to add ‘fromList’ to one of these import lists:
        ‘Data.List.NonEmpty’ (src-lib/Data/HashMap/Strict/Extended.hs:23:1-52)
        ‘Data.HashMap.Strict’ (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46)
   |
30 | fromListOn f = fromList . Prelude.map (\v -> (f v, v))
   |                ^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:34:31: error:
    • Variable not in scope:
        unionWith
          :: (a1 -> a1 -> a1)
             -> HashMap k (NonEmpty v)
             -> HashMap k (NonEmpty v)
             -> HashMap k (NonEmpty v)
    • Perhaps you meant one of these:
        ‘M.unionWith’ (imported from Data.HashMap.Strict),
        ‘unionsWith’ (line 80)
      Perhaps you want to add ‘unionWith’ to the import list
      in the import of ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
34 | unionsAll = F.foldl' (\a b -> unionWith (<>) a (fmap (:|[]) b)) M.empty
   |                               ^^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:70:13: error:
    • Variable not in scope: fromList :: b1 -> HashMap k2 a
    • Perhaps you meant one of these:
        ‘M.fromList’ (imported from Data.HashMap.Strict),
        ‘fromListOn’ (line 30)
      Perhaps you want to add ‘fromList’ to one of these import lists:
        ‘Data.List.NonEmpty’ (src-lib/Data/HashMap/Strict/Extended.hs:23:1-52)
        ‘Data.HashMap.Strict’ (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46)
   |
70 | mapKeys f = fromList . foldrWithKey (\k x xs -> (f k, x) : xs) []
   |             ^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:70:24: error:
    • Variable not in scope:
        foldrWithKey
          :: (k1 -> b0 -> [(k2, b0)] -> [(k2, b0)])
             -> [a0] -> HashMap k1 a -> b1
    • Perhaps you meant one of these:
        ‘M.foldrWithKey’ (imported from Data.HashMap.Strict),
        ‘M.foldlWithKey’ (imported from Data.HashMap.Strict),
        ‘M.foldrWithKey'’ (imported from Data.HashMap.Strict)
      Perhaps you want to add ‘foldrWithKey’ to the import list
      in the import of ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
70 | mapKeys f = fromList . foldrWithKey (\k x xs -> (f k, x) : xs) []
   |                        ^^^^^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:81:15: error:
    • Variable not in scope:
        unionWith
          :: (a -> a -> a) -> HashMap k a -> HashMap k a -> HashMap k a
    • Perhaps you meant one of these:
        ‘M.unionWith’ (imported from Data.HashMap.Strict),
        ‘unionsWith’ (line 80)
      Perhaps you want to add ‘unionWith’ to the import list
      in the import of ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
81 |   = F.foldl' (unionWith f) empty ts
   |               ^^^^^^^^^

src-lib/Data/HashMap/Strict/Extended.hs:81:28: error:
    • Variable not in scope: empty :: HashMap k a
    • Perhaps you meant one of these:
        ‘M.empty’ (imported from Data.HashMap.Strict),
        ‘mempty’ (imported from Prelude)
      Perhaps you want to add ‘empty’ to the import list in the import of
      ‘Data.HashMap.Strict’
      (src-lib/Data/HashMap/Strict/Extended.hs:21:1-46).
   |
81 |   = F.foldl' (unionWith f) empty ts
   |                            ^^^^^
   ```

Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: So I will probably just drop mapKeys again. since its copied in anyway

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: Okay, this should work though:

```diff
- import           Data.HashMap.Strict as M
+ import           Data.HashMap.Strict as M hiding (mapKeys)
```

Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: ok, that worked, now there is another weird error:
```
src-lib/Hasura/Backends/MySQL/Meta.hs:198:22: error:
    parse error on input ‘-- ^ primay key’
    |
198 |         "PRI" -> PRI -- ^ primay key
    |                      ^^^^^^^^^^^^^^^
```

why do all this errors appear? does upstream build with a different haskell version?

Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix:1301]: seems to be a haddock error, I will disable haddock for now

Comment (sternenseemann) [reply pkgs/development/haskell-modules/configuration-common.nix]: Sounds sensible, no clue what upstream is doing…

**Review by sternenseemann — APPROVED** [review]
Seems good, built the following derivations

```
/nix/store/fkwj1iilw1rrcnvdq54yrxpvbvshxqs9-hasura-2.0.5
/nix/store/rnnx0ywc6mmcj7mgbpwbihl9rn1lpwna-odbc-0.2.5
/nix/store/c6bphj5kzw1yl1h9dci23xgd942krlfd-graphql-engine-2.0.5
/nix/store/gjim3pn3ga0cycpk6366cf22x76x2i32-pg-client-0.1.0
```

Leaving it to @cdepillabout to have the final look.

**Review by cdepillabout — APPROVED** [review]
LGTM, thanks for working through this @Lassulus!

File: pkgs/development/haskell-modules/configuration-common.nix
Comment (Lassulus) [reply pkgs/development/haskell-modules/configuration-common.nix]: seems like upstream catched up with haddock :D
https://github.com/hasura/graphql-engine/commit/c1bdc993346c3690ed6fe708cf76959e508b24c6

**Pull request merged**
